File: src/main/java/org/jcodec/containers/mp4/Boxes.java
package org.jcodec.containers.mp4;

import org.jcodec.containers.mp4.boxes.Box;

import js.util.HashMap;
import js.util.Map;

public abstract class Boxes {
    protected final Map<String, Class<? extends Box>> mappings;

    public Boxes() {
        this.mappings = new HashMap<String, Class<? extends Box>>();
    }

    public Class<? extends Box> toClass(String fourcc) {
        return mappings.get(fourcc);
    }

    public void override(String fourcc, Class<? extends Box> cls) {
        mappings.put(fourcc, cls);
    }

    public void clear() {
        mappings.clear();
    }

}

File: src/main/java/org/jcodec/containers/mp4/Boxes.kt
package org.jcodec.containers.mp4

import org.jcodec.containers.mp4.boxes.Box
import java.util.*

abstract class Boxes {
    @JvmField
    protected val mappings: MutableMap<String, Class<out Box?>>
    fun toClass(fourcc: String): Class<out Box?>? {
        return mappings[fourcc]
    }

    fun override(fourcc: String, cls: Class<out Box?>) {
        mappings[fourcc] = cls
    }

    fun clear() {
        mappings.clear()
    }

    init {
        mappings = HashMap()
    }
}